### DISCUSSION TITLE
onCron job example repo

### DISCUSSION THREAD

**Discussion by Arch0125** [MetaMask/snaps#1335] [Q&A]
Hey is there a link to an existing repo which uses onCron job 
Also is the previous bug related to snaps been fixed ?

**Comment by ysfdev2** [comment]
Hey @Arch0125 ,
I was also searching for a cron job example. I found a good example in the test-snaps repo [here](https://github.com/MetaMask/test-snaps/tree/main/packages/cronjob)

**Reply by Arch0125** [reply]
Hey @ysfdev2 thanks for the repo, i tried it out locally, but the dialog wasn't popping up for the given duration, it'll be great if you give detail about how to setup and initiate the cron job

**Reply by ysfdev** [reply]
The dialog is also not being triggered on my example, based on [sample cron job](https://github.com/MetaMask/test-snaps/tree/main/packages/cronjob). 
I think there may be something else missing. I see the scheduler is emitting the onCronJob rpc call but the registered method handler is not being executed. 

Attaching some screenshots for anyone else that may have more context on the issue:

![snap-cronjob-scheduled-executed](https://user-images.githubusercontent.com/13709931/230738145-310c8678-5b7a-4684-9469-d904b2c03c62.jpg)

I see a Method not found error also being logged, not sure if it is related
![method-not-found-error](https://user-images.githubusercontent.com/13709931/230738197-9d21dd3f-c637-4129-aff4-01ad7aa088ce.jpg)

**Reply by ysfdev2** [reply]
Cycling back. I was able to identify the issue/error after seeing the new example from PR #1349.  On my example, I was missing a proper  `snap_dialog`  params `type` value, as it was expected to be CamelCase (e.g: `Alert`, `Comfirmation`, etc). The alert is properly working for me now.
